### PR TITLE
feat(next-queries): update next queries to use the isCurated property

### PR DESCRIPTION
**Pull request by CachedaCodes**
Next Queries now have the property `isCurated`. This PR adds support for the use of that property.

The `NextQuery` component now adds a CSS class to highlight when the NQ is curated. It's also dependant on a new prop set to `false` by default.
The `NextQueries` component also has the same new prop as `NextQuery` that would affect every NQ inside the component.
New tests for the new functionality were also added.

* General efactor of the Next Queries and Next Query.
* Small change in a Related Tag test (out of scope but stumbled upon it)

[EX-5199](https://searchbroker.atlassian.net/browse/EX-5199)

### CODE REVIEW

File: packages/x-components/src/x-modules/next-queries/components/next-query.vue [hunk 1]
```diff
@@ -5,13 +5,15 @@
     :suggestionSelectedEvents="events"
     data-test="next-query"
     feature="next_query"
+    :class="{ 'x-next-query--is-curated': shouldHighlightCurated }"
   >
     <template #default="{ suggestion }">
       <!--
         @slot Next Query content
             @binding {Suggestion} suggestion - Next Query suggestion data
+            @binding {boolean} highlightCurated - True if the curated NQ should be highlighted
```
Comment (pmareke) [inline packages/x-components/src/x-modules/next-queries/components/next-query.vue:14]:
```suggestion
@binding {boolean} shouldHighlightCurated - True if the curated NQ should be highlighted
```

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 2]
```diff
@@ -1,49 +1,87 @@
...
+
+    const findTestDataById = (testDataId: string): WrapperArray<Vue> =>
+      wrapper.findAll(getDataTestSelector(testDataId));
+
+    return {
+      wrapper,
+      nextQueries,
+      findTestDataById,
+      getNextQueryItems() {
+        return findTestDataById('next-query-item');
```
Comment (pmareke) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:55]:
```suggestion
return wrapper.findAll(getDataTestSelector('next-query-item'));
```

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 3]
```diff
@@ -1,49 +1,87 @@
...
+      },
+      {
+        localVue,
+        store,
+        propsData: { nextQueries }
+      }
+    );
+    const wrapper = wrapperTemplate.findComponent(NextQueries);
+
+    const findTestDataById = (testDataId: string): WrapperArray<Vue> =>
```
Comment (pmareke) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:47]:
```suggestion

```

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 4]
```diff
@@ -1,49 +1,87 @@
...
+      wrapper.findAll(getDataTestSelector(testDataId));
```
Comment (pmareke) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:48]:
```suggestion
```

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 5]
```diff
@@ -52,34 +90,45 @@ describe('testing next queries component', () => {
...
+    });
   });
 
-  it('renders a button & a custom Next Query', () => {
-    const wrapperComponent = {
-      template: `
+  it('renders a button, a highlighting icon & a custom Next Query', async () => {
+    const template = `
         <NextQueries>
           <template #suggestion="suggestionScope">
```
Comment (pmareke) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:122]:
```suggestion
 <template #suggestion="{ suggestion, highlightCurated, index }">
```

Also remove the `suggestionScope.` bellow.

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-query.spec.ts [hunk 6]
```diff
@@ -1,45 +1,81 @@
-import { mount } from '@vue/test-utils';
+import { mount, Wrapper } from '@vue/test-utils';
+import { NextQuery } from '@empathyco/x-types';
 import { createNextQueryStub } from '../../../../__stubs__/next-queries-stubs.factory';
 import { getDataTestSelector, installNewXPlugin } from '../../../../__tests__/utils';
 import { getXComponentXModuleName, isXComponent } from '../../../../components/x-component.utils';
 import { WireMetadata } from '../../../../wiring/wiring.types';
-import NextQuery from '../next-query.vue';
+import { default as NextQueryComponent } from '../next-query.vue';
+import { XComponentAPI } from '../../../../plugins/index';
```
Comment (pmareke) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-query.spec.ts:8]:
```suggestion
import { XComponentAPI } from '../../../../plugins/x-plugin.types';
```

File: packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts [hunk 7]
```diff
@@ -1,49 +1,87 @@
...
+import { NextQuery } from '@empathyco/x-types';
 import { getNextQueriesStub } from '../../../../__stubs__';
 import { getDataTestSelector, installNewXPlugin } from '../../../../__tests__/utils';
 import { getXComponentXModuleName, isXComponent } from '../../../../components/x-component.utils';
 import { RootXStoreState } from '../../../../store/store.types';
 import { DeepPartial } from '../../../../utils/types';
 import NextQueries from '../next-queries.vue';
-import NextQuery from '../next-query.vue';
+import { default as NextQueryComponent } from '../next-query.vue';
+import { XPlugin } from '../../../../plugins/index';
```
Comment (pmareke) [inline packages/x-components/src/x-modules/next-queries/components/__tests__/next-queries.spec.ts:12]:
```suggestion
import { XPlugin } from '../../../../plugins/x-plugin';
```

**Review by pmareke — CHANGES_REQUESTED** [review]
Nice, well done!

**Review by diegopf — CHANGES_REQUESTED** [review]

**Review by diegopf — APPROVED** [review]

**Review by pmareke — APPROVED** [review]

**Pull request merged**